### DISCUSSION TITLE
Building rabbitmq-server on windows machine

### DISCUSSION THREAD

**Discussion by Justas293** [rabbitmq/rabbitmq-server#4067] [Other]
Greetings,
I am trying to build a RabbitMQ server on Windows based on instructions from https://www.rabbitmq.com/build-server.html. When I try to `make` or `make run-broker` I get the following error: `Error: No Makefile to build dependency C:/RabbitMQ/deps/jsx`.
I tried `rm -rf deps/jsx`, deleting .erlang.mk, trying the same process on a clean different machine, but that did not help as well.
This is the full error message:
`make[1]: Entering directory '/c/rabbitmq/deps/rabbit_common'
Error: No Makefile to build dependency /c/rabbitmq/deps/jsx.
make[1]: *** [../../erlang.mk:4512: deps] Error 2
make[1]: Leaving directory '/c/rabbitmq/deps/rabbit_common'
make: *** [erlang.mk:4512: deps] Error 2`
I am using the master branch from https://github.com/rabbitmq/rabbitmq-server, running `make` from msys2 bash on a Windows 10 machine.

Does anyone have any ideas about what could be wrong?

**Comment by lukebakken** [comment]
Run using these commands:

```
git clean -xffd
make V=2 FULL=1 2>&1 | tee make.txt
```

*Attach* the output file to your response.

**Reply by Justas293** [reply]
Thank you for your response. Sending the output...
[make.txt](https://github.com/rabbitmq/rabbitmq-server/files/8008081/make.txt)

**Reply by lukebakken** [reply]
```
/bin/sh: line 1: erl: command not found
/bin/sh: line 1: erl: command not found
```

Do you have Erlang installed and in your PATH? If you've installed the Windows Erlang binary, you'll have to add `/c/Program Files/erl-24.2.1/bin` (or some such directory) to your `PATH`. I'm pretty sure msys2 doesn't auto-add Windows PATH items to its own PATH.

Honestly, building RabbitMQ on Windows is such a pain I usually build a windows package using WSL, then unzip the package for testing. Another more recent option is to use Bazel but it has it's own caveats.
